custom/icds_reports/reports/awc_reports.py

```diff
@@ -1107,14 +1107,16 @@ def get_beneficiary_details(case_id, awc_id, selected_month):
     'order', 'reversed_order', 'awc_id'
 ], timeout=30 * 60)
 def get_awc_report_pregnant(order, reversed_order, awc_id):
+    ten_months_ago = datetime.utcnow() - relativedelta(months=10)
```

Comment:
Since all `month` values have the form `YYYY-MM-01` it would be nice to set this date to one where the day ends in 01 because otherwise it will do a scan across a table that is guaranteed to return no results. this can also probably be a date not datetime since the month column is a date.
It also occurs to me that this logic is somewhat wrong and we may need to do two queries. The logic needs to be `all records over the past 10 months for cases where pregnant=1 in the current month` something like `SELECT columns FROM ccs_record_monthly where month > ten_months_ago AND case_id IN (SELECT case_id FROM ccs_record_monthly_view where pregnant=1 and month=current_month)`, but in the django ORM not pseudosql. That might need to be 2 separate queries (one to get the list of ids, then one to get the data for those cases), but might be able to be one with clever `Q objects`

Reply:
It also occurs to me that this report is hard coded to show data from the current month no matter what month was selected. I think the 10 months need to be relative to the selected month in the request, which can be found in the view using the `self.get_settings` method and then passed to all of the pregnant and lactating reports you have created.